src/plugins/lightscript.js

```diff
@@ -121,7 +121,16 @@ pp.parseArrayComprehension = function (node) {
   const loop = this.startNode();
   node.loop = this.parseForStatement(loop);
   this.expect(tt.bracketR);
-  return this.finishNode(node, "ArrayComprehension");
+  node.comprehends = "array";
+  return this.finishNode(node, "Comprehension");
```

Comment:
Yeah, see comment in https://github.com/lightscript/lightscript/issues/6, would rather do code-sharing with helper-functions. Node types should be based around what the thing is, not what makes for easier internals